doc/modules/computational_performance.rst

```diff
@@ -308,6 +308,26 @@ Debian / Ubuntu.
     or upgrade to Python 3.4 which has a new version of ``multiprocessing``
     that should be immune to this problem.
 
+Limiting Working Memory
+-----------------------
+
+Some calculations when implemented using standard numpy vectorized
+operations involve using a large amount of temporary memory.
+As well as potentially exhausting system memory, memory management
+can overwhelm computation time.  Where computations can be performed
```

Comment:
I mean swapping, I suppose. We had repeated reports of terribly slow silhouette calculation, and all the time was spent allocating and then filling memory.

Reply:
Yes, but by chunking NN at least (not sure about `silhouette`) that would still allocate/fill the same amount of memory just sequentially, as far I understand. also related to https://github.com/scikit-learn/scikit-learn/pull/10280#discussion_r160279192

Reply:
Because it does not need to be contiguously addressed and available at the same time it saves swapping time. So you're right, it requires exactly the same number of element accesses, but they can now all be done in main memory where before they had to be swapped to disk.

Reply:
> but they can now all be done in main memory where before they had to be swapped to disk.

OK, in the context of disk swapping I agree. I just have a swap partition of 0MB which means that anything leading to swapping is treated as a memory error, and I tend to forget that it's not a usual behavior.